terminating caduceator causes segmentation fault

Terminating caduceator with ctl+c causes a segmentation fault.
1. start caduceator
2. allow it to run for a few minutes
3. terminate with ctl+c

^C
```
{"caller":"main.go:376","level":"error","msg":"exiting due to signal","signal":"interrupt","ts":"2020-08-26T00:09:30.890421125Z"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xa2b9b3]
goroutine 1 [running]:
github.com/xmidt-org/wrp-listener/webhookClient.(*PeriodicRegisterer).Stop(...)
	/home/centos/git/dev/pkg/mod/github.com/xmidt-org/wrp-listener@v0.1.2/webhookClient/periodicRegisterer.go:74
main.main()
	/home/centos/git/dev/src/caduceator/main.go:387 +0x23d3
```

--- a/caduceator.yaml
+++ b/caduceator.yaml
@@ -151,7 +151,7 @@ vegetaConfig:
   # period is how often events will be sent to Caduceus
   # if frequency is set to 100 and period is set to 1s, then Caduceator will be sending
   # 100 events per second to Caduceus
-  period: "1s"
+  period: "5s"
 
   # connections is the number of idle open connections per target host
   connections: 500
@@ -162,7 +162,7 @@ vegetaConfig:
 
   # maxRoutines is used to limit how long to send events until the number of times a cutoff is reached
   # maxRoutines of 0 means that we will get infinite cutoffs
-  maxRoutines: 1
+  maxRoutines: 0
 
   # url to send events to
   postURL: "http://caduceus:6000/api/v3/notify"
@@ -170,7 +170,7 @@ vegetaConfig:
   # sleepTime is used to slow down the ingestion of events from a consumer's perspective
   # it is the duration of time to sleep before sending a status back to Caduceus upon receiving events
   # a longer sleepTime will result in creating a slower consumer.
-  sleepTime: "45s"
+  sleepTime: "5s"
 
   # client timeout in seconds
   clientTimeout: "5s"
@@ -178,28 +178,32 @@ vegetaConfig:
   # destination of where to send wrpMessage
   wrpMessageDest: "event:device-status/mac:112233445566/offline"
 
-  # vegeta config to simulate a rehash
+  # simulates talaria rehash
   vegetaRehash:
-    # routines is the number of rehashes Caduceus will recieve
-    # if routines is set to 0, no rehashes will take place
     routines: 0
+    period: "1s"
+    frequency: 500
+    connections: 500
+    duration: 0
+    wrpMessageDest: "event:device-status/mac:112233445566/offline"
 
-    # frequency is the number of additional events that will be sent to the
-    # same Caduceus' notify endpoint every period as a rehash.
-    frequency:
-
-    # period is how often a rehash will occur in Caduceus
-    period:
-
-    # connections is the number of idle open connections per target host
-    connections:
-
-    # duration is how long (in minutes) vegeta will be sending events to Caduceus during a rehash period
-    # duration of 0 means vegeta will infinitely keep on attacking caduceus until stopped
-    duration:
-
-    # destination of where to send wrpMessage for rehash events
-    wrpMessageDest:
+  # vegeta config to simulate a rehash
+  # vegetaRehash:
+  #   # routines is the number of rehashes Caduceus will recieve
+  #   # if routines is set to 0, no rehashes will take place
+  #   routines: 0
+  #   # frequency is the number of additional events that will be sent to the
+  #   # same Caduceus' notify endpoint every period as a rehash.
+  #   frequency:
+  #   # period is how often a rehash will occur in Caduceus
+  #   period:
+  #   # connections is the number of idle open connections per target host
+  #   connections:
+  #   # duration is how long (in minutes) vegeta will be sending events to Caduceus during a rehash period
+  #   # duration of 0 means vegeta will infinitely keep on attacking caduceus until stopped
+  #   duration:
+  #   # destination of where to send wrpMessage for rehash events
+  #   wrpMessageDest:
 
 # The following is an example Caduceator vegeta configuration to load test Caduceus
 # of a slow consumer by sending 1000 events per second and only wanting Caduceator
@@ -293,7 +297,7 @@ webhook:
   # registrationInterval provides the time to wait between each registration to
   # the webhook.  If this is set to 0, no registration is done.
   # (Optional)
-  registrationInterval: ""
+  registrationInterval: "3m"
 
   # timeout provides the length of time the request should wait before timing
   # out
@@ -341,16 +345,13 @@ webhook:
     # which endpoints to send to Svalinn.  If the destination of an event
     # matches a regular expression in this list, it is sent to Svalinn
     events: "device-status.*"
-
     # matcher provides regular expressions to match against the event source.
     # (Optional) default is [".*"]
     # matcher:
     #   deviceID: [".*"]
-
   # the below configuration values provide a way to add an Authorization header
   # to the request to the webhook.  If both basic and sat contain empty values,
   # no header is sent.  sat takes priority over basic if both are set.
-
   # basic provides a way to use Basic Authorization when registering to a
   # webhook.  If this value is provided and sat isn't, the following header is
   # added to the registration request:
@@ -358,8 +359,7 @@ webhook:
   # Authorization Basic {basic}
   #
   # (Optional)
-  basic: ""
-
+  basic: "Basic dXNlcjpwYXNz"
   # jwt provides a way to use Bearer Authorization when registering to a
   # webhook.  If the below values are all provided, a request is made to the
   # URL to get the token to be used in the registration request.  The
@@ -368,19 +368,16 @@ webhook:
   # Authorization Bearer {token}
   #
   # (Optional)
-  jwt:
-    # requestHeaders are added to the request for the token.
-    # (Optional)
-    requestHeaders:
-    #   "": ""
-
-    # authURL is the URL to access for the token.
-    authURL: ""
-
-    # timeout is how long the request to get the token will take before
-    # timing out.
-    timeout: "1m"
-
-    # buffer is the length of time before a token expires to get a new token.
-    # (Optional)
-    buffer: "1m"
+  # jwt:
+  #   # requestHeaders are added to the request for the token.
+  #   # (Optional)
+  #   requestHeaders:
+  #     # - "":""
+  #   # authURL is the URL to access for the token.
+  #   authURL: ""
+  #   # timeout is how long the request to get the token will take before
+  #   # timing out.
+  #   timeout: "1m"
+  #   # buffer is the length of time before a token expires to get a new token.
+  #   # (Optional)
+  #   buffer: "1m"

--- a/deploy/docker-compose/docFiles/caduceator.yaml
+++ b/deploy/docker-compose/docFiles/caduceator.yaml
@@ -297,7 +297,7 @@ webhook:
   # registrationInterval provides the time to wait between each registration to
   # the webhook.  If this is set to 0, no registration is done.
   # (Optional)
-  registrationInterval: "3m"
+  registrationInterval: "5m"
 
   # timeout provides the length of time the request should wait before timing
   # out

--- a/deploy/docker-compose/docFiles/caduceus_spruce.yaml
+++ b/deploy/docker-compose/docFiles/caduceus_spruce.yaml
@@ -242,7 +242,7 @@ webhookStore:
   bucket: "webhooks-caduceus"
   address: (( grab $ARGUS_HOST || "argus:6600" ))
   pullInterval: "2s"
-  defaultTTL: 120
+  defaultTTL: 300
   auth:
     basic: (( grab $AUTH_HEADER || "Basic dXNlcjpwYXNz" ))
 ########################################

--- a/go.mod
+++ b/go.mod
@@ -18,5 +18,5 @@ require (
 	github.com/xmidt-org/bascule v0.8.1
 	github.com/xmidt-org/webpa-common v1.6.3
 	github.com/xmidt-org/wrp-go v1.3.4
-	github.com/xmidt-org/wrp-listener v0.2.1
+	github.com/xmidt-org/wrp-listener v0.2.2-0.20201008164425-8cc15715a0bb
 )

--- a/main.go
+++ b/main.go
@@ -315,7 +315,12 @@ func main() {
 			os.Exit(1)
 		}
 
-		periodicRegisterer := webhookClient.NewPeriodicRegisterer(registerer, config.Webhook.RegistrationInterval, logger, metricsRegistry)
+		periodicRegisterer, err := webhookClient.NewPeriodicRegisterer(registerer, config.Webhook.RegistrationInterval, logger, metricsRegistry)
+
+		if err != nil {
+			logging.Error(logger).Log(logging.MessageKey(), "failed to setup periodic registerer", logging.ErrorKey(), err.Error())
+			os.Exit(1)
+		}
 		periodicRegisterersList = append(periodicRegisterersList, periodicRegisterer)
 
 		periodicRegisterer.Start()
@@ -390,7 +395,10 @@ func main() {
 
 	metrics.Close()
 	for i := 0; i < len(periodicRegisterersList); i++ {
-		periodicRegisterersList[i].Stop()
+		if periodicRegisterersList[i] != nil {
+			periodicRegisterersList[i].Stop()
+		}
+
 	}
 	close(shutdown)
 	waitGroup.Wait()

